generator/schemas/gen.go

```diff
@@ -131,12 +147,54 @@ func (g Generator) Generate(ctx *genall.GenerationContext) error {
 				root.AddError(fmt.Errorf("In order to generate Json schemas from the K8S API, you should provide a valid semver-compatible devfile version in the +devfile:jsonschema:version comment marker of the K8S API package (in the doc.go file)"))
 				return nil
 			}
+			forRoot.devfileSchemaVersion = devfileSchemaVersion
 		default:
 			root.AddError(fmt.Errorf("In order to generate Json schemas from the K8S API, you should annotate the K8S API package (in the doc.go file) with the +devfile:jsonschema:version comment marker"))
 			return nil
 		}
 
-		for _, typeToProcess := range jsonschemaRequested {
+		switch groupName := packageMarkers.Get("groupName").(type) {
+		case string:
+			forRoot.groupName = groupName
+		default:
+			root.AddError(fmt.Errorf("The package should have a valid 'groupName' annotation"))
+			return nil
+		}
+		groupVersion := schema.GroupVersion{
+			Group:   forRoot.groupName,
+			Version: forRoot.version,
+		}
+
+		apiVersionsByAPIGroup[groupVersion.Group] = append(apiVersionsByAPIGroup[groupVersion.Group], groupVersion.Version)
+		schemaVersionsByGV[groupVersion] = forRoot.devfileSchemaVersion
+		packageByGV[groupVersion] = root
+		toGenerateByPackage[root] = forRoot
+	}
+
+	for groupName, apiVersions := range apiVersionsByAPIGroup {
+		genutils.SortKubeLikeVersion(apiVersions)
+
+		var currentSchemaVersion *semver.Version
+		var currentAPIVersion string
+		for _, apiVersion := range apiVersions {
+			groupVersion := schema.GroupVersion{Group: groupName, Version: apiVersion}
+			schemaVersion := schemaVersionsByGV[groupVersion]
+			if currentSchemaVersion != nil && schemaVersion != nil {
+				if schemaVersion.Compare(*currentSchemaVersion) <= 0 {
+					packageByGV[groupVersion].AddError(fmt.Errorf("The schema versions should be incremented on each increment of the corresponding K8S apiVersion.\n" +
+						"This is not the case in the '" + groupName + "' API group:\n" +
+						"  '" + currentAPIVersion + "' K8S apiVersion => '" + currentSchemaVersion.String() + "' Json schema version\n" +
+						"  '" + apiVersion + "' K8S apiVersion => '" + schemaVersion.String() + "' Json schema version\n"))
```

Comment:
If we're using `fmt.Errorf` we should use formatting rather than string concatenation:
```suggestion
					packageByGV[groupVersion].AddError(fmt.Errorf("The schema versions should be incremented on each increment of the corresponding K8S apiVersion.\n" +
						"This is not the case in the "%s' API group:\n" +
						"  '%s' K8S apiVersion => '%s' Json schema version\n" +
						"  '%s' K8S apiVersion => '%s' Json schema version\n", groupName, currentAPIVersion, currentSchemaVersion.String(), apiVersion, schemaVersion.String()))
```
Consider also using a raw string to avoid having to include `\n" +`